src/app/src/hooks/useDownloads.ts

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+
+import { downloadResultsCsv, downloadResultsJson } from '../utils/api/downloads';
+import { useToast } from './useToast';
+export function downloadBlob(blob: Blob, fileName: string) {
+  const url = URL.createObjectURL(blob);
+  const link = document.createElement('a');
+  link.href = url;
+  link.download = fileName;
+  document.body.appendChild(link);
+  link.click();
+  document.body.removeChild(link);
+  URL.revokeObjectURL(url);
+}
+
+interface UseDownloadOptions {
+  onSuccess?: (fileName: string) => void;
+  onError?: (error: Error) => void;
+}
+
+type DownloadFormat = 'csv' | 'json';
+
+/**
+ * Generic hook for downloading evaluation results
+ * @param format The download format (csv or json)
+ * @param options Callbacks for success and error events
+ * @returns Download function and loading state
+ */
+export function useDownload(format: DownloadFormat, options?: UseDownloadOptions) {
```

Comment:
The hook mixes patterns (returning state, firing callbacks, showing toasts, and throwing errors) which makes it hard to use predictably. Please streamline to a declarative, unidirectional contract that returns a stable {isLoading, error, fileName, download} and keep side-effects (like toasts) in the consumer.